src/app/[locale]/incident/add/components/MapDialog.tsx

```diff
@@ -115,20 +160,36 @@ const MapDialog = ({
 
   // Handle click on map, setIsMapSelected to true
   // TODO: Reset selectedFeatures if click was right on map? (open for discussion)
-  const handleMapClick = (event: MapLayerMouseEvent) => {
+  const handleMapClick = async (event: MapLayerMouseEvent) => {
     updatePosition(event.lngLat.lat, event.lngLat.lng)
-
     setIsMapSelected(true)
+    const address = await getNewSelectedAddress(
+      event.lngLat.lat,
+      event.lngLat.lng
+    )
+
+    updateForm({
+      ...formState,
+      address: address,
+    })
   }
 
   // Handle click on feature marker, set selectedFeatures and show error if maxNumberOfAssets is reached
-  const handleFeatureMarkerClick = (event: MarkerEvent, feature: Feature) => {
+  const handleFeatureMarkerClick = async (
```

Comment:
Wanneer je in stap 2 een adres hebt geselecteerd, vervolgens het formulier doorwerkt en vanuit stap 4 terug navigeert naar stap 2 en de kaart opent is de marker niet meer te zien.

Reply:
gaat ook fout bij:

stap 1: lantaarn kapot
stap 2: vul adres in input veld -> verschijnt marker -> klik op kaart om objecten te selecteren -> marker weg